codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoder.java

```diff
@@ -401,6 +401,15 @@ public void onHeadersRead(ChannelHandlerContext ctx, int streamId, Http2Headers
                                 "Multiple content-length headers received");
                     }
                 }
+            } else {
+                // Need to check trailers don't contain pseudo headers. According to RFC 9113
+                // Trailers MUST NOT include pseudo-header fields (Section 8.3).
+                for (CharSequence name : headers.names()) {
```

Comment:
headers.names() copies names into new LinkedHashSet - for iteration It is sufficient to use Http2Headers.iterator() which does not do redundant copy (possibly guarded by isEmpty() check returning Collections.emptyIterator())

Reply:
Thanks for the comment! Makes sense, I'll switch it up to use the iterator. From what I could see in the headers code, I don't think it's necessary to add an `isEmpty()` check since it should be covered by the implementation but doesn't hurt to have it so I'll add it in too. Should have that pushed in a bit.